launch_ros/launch_ros/substitutions/find_package.py

```diff
@@ -39,10 +41,16 @@ class FindPackage(Substitution):
         not found during substitution.
     """
 
-    def __init__(self, package: SomeSubstitutionsType) -> None:
+    def __init__(
+        self,
+        package: SomeSubstitutionsType,
+        *,
+        find_package_func: Callable[[str], str] = get_package_prefix
+    ) -> None:
         """Constructor."""
         super().__init__()
         self.__package = normalize_to_list_of_substitutions(package)
+        self.__find_package_func = find_package_func
```

Comment:
@jacobperron out of curiosity, why pass in a function as opposed to regular method overriding by subclasses?

Reply:
I hadn't thought of that. I presume you mean defining a new method `find_package` that can be overridden by subclasses. Makes sense to me.

What do you think about turning `FindPackage` into an abstract class with a `find_package` method that needs to be overridden by subclasses? Then we can add a `FindPackagePrefix` (and leaves open the possibility for future subclasses like `FindPackageBin`).

Reply:
Sure, that makes sense to me.

Reply:
See 569ef58